plugins/woocommerce/tests/legacy/unit-tests/rest-api/Tests/Version2/system-status.php

```diff
@@ -17,18 +17,52 @@ class WC_Tests_REST_System_Status_V2 extends WC_REST_Unit_Test_Case {
 	use ArraySubsetAsserts;
 
 	/**
-	 * Setup our test server.
+	 * User variable.
+	 *
+	 * @var int
+	 */
+	protected static int $administrator_user;
+
+	/**
+	 * User variable.
+	 *
+	 * @var int
+	 */
+	protected static int $no_user = 0;
+
+	/**
+	 * Setup once before running tests.
+	 *
+	 * @param object $factory Factory object.
 	 */
-	public function setUp(): void {
-		parent::setUp();
-		$this->endpoint = new WC_REST_System_Status_Controller();
-		$this->user     = $this->factory->user->create(
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$administrator_user = $factory->user->create(
 			array(
 				'role' => 'administrator',
 			)
 		);
 	}
 
+	/**
+	 * Fetches the System Status Report data and caches it.
+	 * @param  int $user The ID of a WordPress user to switch to before fetching the data.
+	 * @return Array An array of the data returned by the System Status Report endpoint.
+	 */
+	private function fetch_or_get_system_status_data_for_user( int $user ) {
+		if ( $user < 0 ) {
+			return null;
+		}
+
+		static $system_status_data = array();
+		if ( ! array_key_exists( 'user' . $user, $system_status_data ) ) {
```

Comment:
This isn't a blocker, but, why th use of `array_key_exists` instead of `isset`?

Reply:
Good call -- looked up the difference and found `isset` is probably a better fit here. 👍